### PR TITLE
PERF: only apply nanops rowwise optimization for narrow arrows

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
See the timings in https://github.com/pandas-dev/pandas/pull/43311#issuecomment-974891737

(this is a regression compared to previous release mostly when using ArrayManager (and thus doesn't need a whatsnew) and is covered by the FrameOps benchmarks)

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -463,6 +463,7 @@ def newfunc(values: np.ndarray, *, axis: int | None = None, **kwargs):
             axis == 1
             and values.ndim == 2
             and values.flags["C_CONTIGUOUS"]
+            and (values.shape[1] / values.shape[0]) > 1000
```
Comment (jreback) [inline pandas/core/nanops.py:466]: can you add some comments about why these things are needed (e.g. for performance), can ref this issue as well.

Comment (jorisvandenbossche) [reply pandas/core/nanops.py:470]: Added a comment with reference to my timings and update the docstring

**Comment by jorisvandenbossche** [comment]
Verified that the remaining failure is only an unrelated ResourceWarning

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Comment by jbrockmendel** [comment]
LGTM.  merits an asv?

**Comment by jorisvandenbossche** [comment]
> LGTM. merits an asv?

It gave a huge slowdown in the current ASVs when running with ArrayManager, so in that way it's already captured. 

We could add one for wide dataframes as well, but you already need an extreme shape (like 10, 1000), so I personally don't find that a very interesting one to add.

**Comment by jbrockmendel** [comment]
> so in that way it's already captured

id be more comfortable with one that always runs

**Pull request merged**